java/shims/src/java/org/apache/orc/impl/ZeroCopyShims.java

```diff
@@ -69,14 +71,30 @@ public ByteBuffer readBuffer(int maxLength, boolean verifyChecksums)
       if (verifyChecksums) {
         options = CHECK_SUM;
       }
-      return this.in.read(this.pool, maxLength, options);
+
+      ByteBuffer bb = this.in.read(this.pool, maxLength, options);
+      readBuffers.put(bb, null);
+      return bb;
     }
 
+    /**
+     * @deprecated Use {@link #releaseAllBuffers()} instead. This method was
+     * incorrectly added and shouldn't be used anymore.
+     */
+    @Deprecated
     @Override
     public void releaseBuffer(ByteBuffer buffer) {
       this.in.releaseBuffer(buffer);
     }
 
+    @Override
+    public void releaseAllBuffers() {
+      readBuffers.forEach((k, v) -> {
+        this.in.releaseBuffer(k);
+      });
+      readBuffers.clear();
+    }
+
     @Override
     public void close() throws IOException {
```

Comment:
I'm wondering if we need to invoke `releaseAllBuffers` in `close()` too because `readBuffers` is a newly added structure.

Reply:
Thanks for your review, I have added it. I think it make sense from the point of code design. 
BTW, `releaseAllBuffers ` is a step of `StripePlanner#clearStreams`, we always `releaseAllBuffers` before `close()` now.